src/framework/mlt_frame.h

```diff
@@ -146,6 +146,11 @@ extern mlt_frame mlt_frame_clone( mlt_frame self, int is_deep );
 /* convenience functions */
 extern int mlt_sample_calculator( float fps, int frequency, int64_t position );
 extern int64_t mlt_sample_calculator_to_now( float fps, int frequency, int64_t position );
+extern void mlt_audio_copy( void* dst, void* src, mlt_audio_format format, int channels, int samples, int src_start, int dst_start, int src_max, int dst_max );
+extern int mlt_audio_plane_count( mlt_audio_format format, int channels );
+extern int mlt_audio_plane_size( mlt_audio_format format, int samples, int channels );
+extern void mlt_audio_collapse_channels( mlt_audio_format format, int channels, int allocated_samples, int used_samples, void* buffer );
+extern void mlt_audio_reverse( mlt_audio_format format, int channels, int samples, void* buffer );
```

Comment:
After I copy/paste a function for the third time, I feel like it should be moved into the framework. These could all be moved into a new "audio" module if we think the frame file is too cluttered. I think it would be nice if we added an "audio" class to encapsulate format, channels, samples and buffer into a single struct with these (and other) convenience functions.

Reply:
Are you thinking about trying to replace the various audio properties in mlt_frame with a single struct? If so, that might mean a lot of changes and potentially with some apps. Otherwise, for now, I am content to have mlt_audio.[ch] and mlt_image.[ch] to move functions out of mlt_frame.[ch].

Reply:
Yes. That is what I was thinking. If we make such a change, I would volunteer to change all the code. But I expect it would also break some applications or 3rd party modules. Perhaps there is a way to keep the current interface and add a new one. The current interface application could be a backwards compatible wrapper for the new interface. I would be willing to submit a proposal for this if you are interested.

I will move the audio/image functions into separate files. That would be a step in that direction.